zero/ifc/text/format.cppm

```diff
@@ -0,0 +1,88 @@
+/**
+* Provides functions and enums for string formatting and styling
+*/
+
+export module format;
+
+import std;
+
+template<typename T>
+std::string stringify(const T& t) {
+    std::ostringstream oss;
+    oss << t;
+    return oss.str();
+}
+
+export namespace zero::fmt {
+
+    enum class Color {
+        DEFAULT,  // Default console color
+        RED = 196,
+        ORANGE = 202,
+        YELLOW = 226,
+        GREEN = 46,
+        BLUE = 21,
+        PURPLE = 93,
+        CYAN = 51,
+        WHITE = 15,
+        GREY = 244,
+        BLACK = 16
+    };
+
+    enum class Modifier {
+        BOLD = 1,
+        FAINT = 2,
+        ITALIC = 3,
+        UNDERLINE = 4,
+        BLINK = 5,
+        REVERSE = 7,
+        HIDDEN = 8
+    };
+
+    template<typename... Args>
+    std::string formatter(const std::string& format, Args... args) {
+        std::deque<std::string> argStrings{ stringify(args)... };
+        std::string result;
+        bool escape_next = false;
+
+        for (size_t i = 0; i < format.size(); ++i) {
+            if (format[i] == '\\' && !escape_next) {
+                escape_next = true;
+            } else if (format[i] == '{' && i + 1 < format.size() && format[i + 1] == '}' && !escape_next) {
+                if (argStrings.empty())
+                    throw std::runtime_error("Not enough arguments provided");
```

Comment:
As we've discussed before, the Zero codebase must not support non-exception free code (only, for now, in the TestSuite is allowed, but just because does not throws directly to the user code, just for tracking internal testing features). Even tho, I am happy with this implementation for now, since getting rid of the exceptions means moving the check for that the number of arguments matches the number of placeholders. So just take it in consideration for future implementations